pkg/report/datafeeds/id.go

```diff
@@ -90,3 +92,33 @@ func GetDecimals(dataType uint8) (uint8, bool) {
 	// Else if the data type is not a number
 	return 0, false
 }
+
+func (rm Metadata) Encode() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	err := binary.Write(buf, binary.BigEndian, rm)
+	if err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+func (rm Metadata) Length() int {
+	bytes, err := rm.Encode()
+	if err != nil {
+		return 0
+	}
+	return len(bytes)
+}
+
+// Note: This should be a shared type that the OCR3 package validates as well
+type Metadata struct {
```

Comment:
I'd prefer if this goes to a specific `metadata.go` file with a comment and a TODO to clean up and align with the common type.